distributed/worker.py

```diff
@@ -3395,17 +3393,17 @@ def done_event():
                     ts.who_has.discard(worker)
                     self.has_what[worker].discard(ts.key)
                     self.log.append((d, "missing-dep", stimulus_id, time()))
-                    self.batched_stream.send(
-                        {
-                            "op": "missing-data",
-                            "errant_worker": worker,
-                            "key": d,
-                            "stimulus_id": stimulus_id,
-                        }
+                    instructions.append(
+                        MissingDataMsg(
+                            key=d,
+                            errant_worker=worker,
+                            stimulus_id=stimulus_id,
+                        )
                     )
                     recommendations[ts] = "fetch"
             del data, response
             self.transitions(recommendations, stimulus_id=stimulus_id)
+            self._handle_instructions(instructions)
```

Comment:
This `MissingDataMsg` will now come _after_ any messages generated by `transitions`. I think that shouldn't matter, but want to double check.

Reply:
I would be seriously worried if order mattered

Reply:
Ordering does sometimes matter and I suggest to not reject such a possibility just because it _would_ be bad design. The scheduler _should_ be able to handle all possible kind of orderings but this can still introduce subtle and unintended changes in behavior and I would advise caution either way